spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -117,7 +119,7 @@ public class SparkTable
           .build();
 
   private final Table icebergTable;
-  private final Long snapshotId;
+  private Long snapshotId;
```

Comment:
Do we have to remove the final keyword here?

Reply:
Because we have two different constructors, `snapshotId` is not final be unless we change the logic, specifically the below part.
```java
    this(icebergTable, refreshEagerly); // <- here sets snapshotId
    this.branch = branch;
    Snapshot snapshot = icebergTable.snapshot(branch);
    ValidationException.check(
        branch == null || SnapshotRef.MAIN_BRANCH.equals(branch) || snapshot != null,
        "Cannot use branch (does not exist): %s",
        branch);
    this.snapshotId = snapshot.snapshotId(); // <- try to set snapshotId again.
```